Crash on typing into AI panel

### Summary

<!-- Please insert a one line summary of the issue below -->

<!-- Include all steps necessary to reproduce from a clean Zed installation. Be verbose -->
Steps to trigger the problem:
1. Type text into the AI panel
2. Crash 



### Zed Version and System Specs

Zed: v0.174.6 (Zed) 
OS: macOS 15.2.0
Memory: 16 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary>

<!-- Click below this line and paste or drag-and-drop your log-->
```
2025-02-22T12:27:13.208901Z [INFO] attempting to start language server "dockerfile-language-server", path: "/Users/sakibulislam/Github/learn-framer-university", id: 10
2025-02-22T12:27:14.225421Z [INFO] starting language server process. binary path: "/opt/homebrew/bin/node", working directory: "/Users/sakibulislam/Github/learn-framer-university", args: ["/Users/sakibulislam/Library/Application Support/Zed/extensions/work/dockerfile/node_modules/dockerfile-language-server-nodejs/bin/docker-langserver", "--stdio"]
2025-02-22T12:27:21.196075Z [ERROR] edit prediction failed

Caused by:
    error predicting edits.
    Status: 500
    Body: Invalid output: <|editable_region_start|>
    ENV NEXT_PUBLIC_APP_URL=${NEXT_PUBLIC_APP_URL}
    
    LABEL fly_launch_runtime="Next.js"
    
    # Next.js app lives here
    WORKDIR /app
    
    # Set production environment
    ENV NODE_ENV="production"
    
    # Throw-away build stage to reduce size of final image
    FROM base AS build
    
    # Install packages needed to build node modules
    RUN apt-get update -qq && \
        apt-get install --no-install-recommends -y build-essential node-gyp pkg-config python-is-python3
    
    # Install node modules
    COPY package-lock.json package.json ./
    RUN npm ci --include=dev
    
    # Copy application code
    COPY . .
    
    # Build application
    RUN npx next build
    
    # Remove development dependencies
    RUN npm prune --omit=dev
    
    
    # Final stage for app image
    FROM base
    
    # Copy standalone build output
    COPY --from=build /app/.next/standalone /app
    COPY --from=build /app/.next/static /app/.next/static
    COPY --from=build /app/public /app/public
    
    # Create and copy entrypoint script
    COPY docker-entrypoint.js /app/docker-entrypoint.js
    RUN chmod +x /app/docker-entrypoint.js
    
    # Start the server by default, this can be overwritten at runtime
    EXPOSE 3000
    
    # Use the entrypoint script
    CMD ["node", "server.js"]
    
    # Add a healthcheck to ensure the server is up and running
    HEALTHCHECK --interval=30s --timeout=5s --retries=3 CMD curl -f http://localhost:3000/health || exit 1
    
    ```
    
2025-02-22T12:27:23.092027Z [ERROR] edit prediction failed

Caused by:
    error predicting edits.
    Status: 500
    Body: Invalid output: <|editable_region_start|>
    ENV NEXT_PUBLIC_APP_URL=${NEXT_PUBLIC_APP_URL}
    
    LABEL fly_launch_runtime="Next.js"
    
    # Next.js app lives here
    WORKDIR /app
    
    # Set production environment
    ENV NODE_ENV="production"
    
    # Throw-away build stage to reduce size of final image
    FROM base AS build
    
    # Install packages needed to build node modules
    RUN apt-get update -qq && \
        apt-get install --no-install-recommends -y build-essential node-gyp pkg-config python-is-python3
    
    # Install node modules
    COPY package-lock.json package.json ./
    RUN npm ci --include=dev
    
    # Copy application code
    COPY . .
    
    # Build application
    RUN npx next build
    
    # Remove development dependencies
    RUN npm prune --omit=dev
    
    
    # Final stage for app image
    FROM base
    
    # Copy standalone build output
    COPY --from=build /app/.next/standalone /app
    COPY --from=build /app/.next/static /app/.next/static
    COPY --from=build /app/public /app/public
    
    # Create and copy entrypoint script
    COPY docker-entrypoint.js /app/docker-entrypoint.js
    RUN chmod +x /app/docker-entrypoint.js
    
    # Start the server by default, this can be overwritten at runtime
    EXPOSE 3000
    
    # Use the entrypoint script
    CMD ["node", "server.js"]
    
    ```
    
2025-02-22T12:27:46.025295Z [ERROR] edit prediction failed

Caused by:
    error predicting edits.
    Status: 500
    Body: Invalid output: <|editable_region_start|>
    COPY --from=build /app/public /app/public
    
    # Create and copy entrypoint script
    COPY docker-entrypoint.js /app/docker-entrypoint.js
    RUN chmod +x /app/docker-entrypoint.js
    
    # Start the server by default, this can be overwritten at runtime
    EXPOSE 3000
    
    # Use the entrypoint script
    CMD ["node", "server.js"]
    ``
    
    # Final stage for app image
    FROM base
    
    # Copy standalone build output
    COPY --from=build /app/.next/standalone /app
    COPY --from=build /app/.next/static /app/.next/static
    <|editable_region_start|>
    COPY --from=build /app/public /app/public
    
    # Create and copy entrypoint script
    COPY docker-entrypoint.js /app/docker-entrypoint.js
    RUN chmod +x /app/docker-entrypoint.js
    
    # Start the server by default, this can be overwritten at runtime
    EXPOSE 3000
    
    # Use the entrypoint script
    CMD ["node", "server.js"]
    ``
    
    # Final stage for app image
    FROM base
    
    # Copy standalone build output
    COPY --from=build /app/.next/standalone /app
    COPY --from=build /app/.next/static /app/.next/static
    <|editable_region_start|>
    COPY --from=build /app/public /app/public
    
    # Create and copy entrypoint script
    COPY docker-entrypoint.js /app/docker-entrypoint.js
    RUN chmod +x /app/docker-entrypoint.js
    
    # Start the server by default, this can be overwritten at runtime
    EXPOSE 3000
    
    # Use the entrypoint script
    CMD ["node", "server.js"]
    ``
    
    # Final stage for app image
    FROM base
    
    # Copy standalone build output
    COPY --from=build /app/.next/standalone /app
    COPY --from=build /app/.next/static /app/.next/static
    <|editable_region_start|>
    COPY --from=build /app/public /app/public
    
    # Create and copy entrypoint script
    COPY docker-entrypoint.js /app/docker-entrypoint.js
    RUN chmod +x /app/docker-entrypoint.js
    
    # Start the server by default, this can be overwritten at runtime
    EXPOSE 3000
    
    # Use the entrypoint script
    CMD ["node", "server.js"]
    ``
    
    # Final stage for app image
    FROM base
    
    # Copy standalone build output
    COPY --from=build /app/.next/standalone /app
    COPY --from=build /app/.next/static /app/.next/static
    <|editable_region_start|>
    COPY --from=build /app/public /app/public
    
    # Create and copy entrypoint script
    COPY docker-entrypoint.js /app/docker-entrypoint.js
    RUN chmod +x /app/docker-entrypoint.js
    
    # Start the server by default, this can be overwritten at runtime
    EXPOSE 3000
    
    # Use the entrypoint script
    CMD ["node", "server.js"]
    ``
    
    # Final stage for app image
    FROM base
    
    # Copy standalone build output
    COPY --from=build /app/.next/standalone /app
    COPY --from=build /app/.next/static /app/.next/static
    <|editable_region_start|>
    COPY --from=build /app/public /app/public
    
    # Create and copy entrypoint script
    COPY docker-entrypoint.js /app/docker-entrypoint.js
    RUN chmod +x /app/docker-entrypoint.js
    
    # Start the server by default, this can be overwritten at runtime
    EXPOSE 3000
    
    # Use the entrypoint script
    CMD ["node", "server.js"]
    ``
    
    # Final stage for app image
    FROM base
    
    # Copy standalone build output
    COPY --from=build /app/.next/standalone /app
    COPY --from=build /app/.next/static /app/.next/static
    <|editable_region_start|>
    COPY --from=build /app/public /app/public
    
    # Create and copy entrypoint script
    COPY docker-entrypoint.js /app/docker-entrypoint.js
    RUN chmod +x /app/docker-entrypoint.js
    
    # Start the server by default, this can be overwritten at runtime
    EXPOSE 3000
    
    # Use the entrypoint script
    CMD ["node", "server.js"]
    ``
    
    # Final stage for app image
    FROM base
    
    # Copy standalone build output
    COPY --from=build /app/.next/standalone /app
    COPY --from=build /app/.next/static /app/.next/static
    <|editable_region_start|>
    COPY --from=build /app/public /app/public
    
    # Create and copy entrypoint script
    COPY docker-entrypoint.js /app/docker-entrypoint.js
    RUN chmod +x /app/docker-entrypoint.js
    
    # Start the server by default, this can be overwritten at runtime
    EXPOSE 3000
    
    
2025-02-22T12:27:46.358739Z [ERROR] edit prediction failed

Caused by:
    error predicting edits.
    Status: 500
    Body: Invalid output: <|editable_region_start|>
    COPY --from=build /app/public /app/public
    
    # Create and copy entrypoint script
    COPY docker-entrypoint.js /app/docker-entrypoint.js
    RUN chmod +x /app/docker-entrypoint.js
    
    # Start the server by default, this can be overwritten at runtime
    EXPOSE 3000
    
    # Use the entrypoint script
    CMD ["node", "server.js"]
    ``
    
    # Final stage for app image
    FROM base
    
    # Copy standalone build output
    COPY --from=build /app/.next/standalone /app
    COPY --from=build /app/.next/static /app/.next/static
    <|editable_region_start|>
    COPY --from=build /app/public /app/public
    
    # Create and copy entrypoint script
    COPY docker-entrypoint.js /app/docker-entrypoint.js
    RUN chmod +x /app/docker-entrypoint.js
    
    # Start the server by default, this can be overwritten at runtime
    EXPOSE 3000
    
    # Use the entrypoint script
    CMD ["node", "server.js"]
    ``
    
    # Final stage for app image
    FROM base
    
    # Copy standalone build output
    COPY --from=build /app/.next/standalone /app
    COPY --from=build /app/.next/static /app/.next/static
    <|editable_region_start|>
    COPY --from=build /app/public /app/public
    
    # Create and copy entrypoint script
    COPY docker-entrypoint.js /app/docker-entrypoint.js
    RUN chmod +x /app/docker-entrypoint.js
    
    # Start the server by default, this can be overwritten at runtime
    EXPOSE 3000
    
    # Use the entrypoint script
    CMD ["node", "server.js"]
    ``
    
    # Final stage for app image
    FROM base
    
    # Copy standalone build output
    COPY --from=build /app/.next/standalone /app
    COPY --from=build /app/.next/static /app/.next/static
    <|editable_region_start|>
    COPY --from=build /app/public /app/public
    
    # Create and copy entrypoint script
    COPY docker-entrypoint.js /app/docker-entrypoint.js
    RUN chmod +x /app/docker-entrypoint.js
    
    # Start the server by default, this can be overwritten at runtime
    EXPOSE 3000
    
    # Use the entrypoint script
    CMD ["node", "server.js"]
    ``
    
    # Final stage for app image
    FROM base
    
    # Copy standalone build output
    COPY --from=build /app/.next/standalone /app
    COPY --from=build /app/.next/static /app/.next/static
    <|editable_region_start|>
    COPY --from=build /app/public /app/public
    
    # Create and copy entrypoint script
    COPY docker-entrypoint.js /app/docker-entrypoint.js
    RUN chmod +x /app/docker-entrypoint.js
    
    # Start the server by default, this can be overwritten at runtime
    EXPOSE 3000
    
    # Use the entrypoint script
    CMD ["node", "server.js"]
    ``
    
    # Final stage for app image
    FROM base
    
    # Copy standalone build output
    COPY --from=build /app/.next/standalone /app
    COPY --from=build /app/.next/static /app/.next/static
    <|editable_region_start|>
    COPY --from=build /app/public /app/public
    
    # Create and copy entrypoint script
    COPY docker-entrypoint.js /app/docker-entrypoint.js
    RUN chmod +x /app/docker-entrypoint.js
    
    # Start the server by default, this can be overwritten at runtime
    EXPOSE 3000
    
    # Use the entrypoint script
    CMD ["node", "server.js"]
    ``
    
    # Final stage for app image
    FROM base
    
    # Copy standalone build output
    COPY --from=build /app/.next/standalone /app
    COPY --from=build /app/.next/static /app/.next/static
    <|editable_region_start|>
    COPY --from=build /app/public /app/public
    
    # Create and copy entrypoint script
    COPY docker-entrypoint.js /app/docker-entrypoint.js
    RUN chmod +x /app/docker-entrypoint.js
    
    # Start the server by default, this can be overwritten at runtime
    EXPOSE 3000
    
    # Use the entrypoint script
    CMD ["node", "server.js"]
    ``
    
    # Final stage for app image
    FROM base
    
    # Copy standalone build output
    COPY --from=build /app/.next/standalone /app
    COPY --from=build /app/.next/static /app/.next/static
    <|editable_region_start|>
    COPY --from=build /app/public /app/public
    
    # Create and copy entrypoint script
    COPY docker-entrypoint.js /app/docker-entrypoint.js
    RUN chmod +x /app/docker-entrypoint.js
    
    # Start the server by default, this can be overwritten at runtime
    EXPOSE 3000
    
    
2025-02-22T12:27:47.041888Z [ERROR] edit prediction failed

Caused by:
    error predicting edits.
    Status: 500
    Body: Invalid output: <|editable_region_start|>
    CMD ["node", "server.js"]
    ```
    
    ### Explanation of Changes:
    1. **Install pnpm**: Added `RUN npm install -g pnpm` to install pnpm globally.
    2. **Install node modules**: Replaced `COPY package-lock.json package.json ./` with `COPY pnpm-lock.yaml ./` and `COPY package.json ./` to copy the pnpm lock file and package.json.
    3. **Install dependencies**: Replaced `RUN npm ci --include=dev` with `RUN pnpm install --frozen-lockfile` to install dependencies using pnpm.
    4. **Build application**: Replaced `RUN npx next build` with `RUN pnpm exec next build` to build the application using pnpm.
    5. **Prune development dependencies**: Replaced `RUN npm prune --omit=dev` with `RUN pnpm prune --prod` to remove development dependencies using pnpm.
    
    With these changes, your Dockerfile is now set up to use pnpm instead of npm.
    and this?
    <|editable_region_start|>
    CMD ["node", "server.js"]
    ```
    
    ### Explanation of Changes:
    1. **Install pnpm**: Added `RUN npm install -g pnpm` to install pnpm globally.
    2. **Install node modules**: Replaced `COPY package-lock.json package.json ./` with `COPY pnpm-lock.yaml ./` and `COPY package.json ./` to copy the pnpm lock file and package.json.
    3. **Install dependencies**: Replaced `RUN npm ci --include=dev` with `RUN pnpm install --frozen-lockfile` to install dependencies using pnpm.
    4. **Build application**: Replaced `RUN npx next build` with `RUN pnpm exec next build` to build the application using pnpm.
    5. **Prune development dependencies**: Replaced `RUN npm prune --omit=dev` with `RUN pnpm prune --prod` to remove development dependencies using pnpm.
    
    With these changes, your Dockerfile is now set up to use pnpm instead of npm.
    and this?
    <|editable_region_start|>
    CMD ["node", "server.js"]
    ```
    
    ### Explanation of Changes:
    1. **Install pnpm**: Added `RUN npm install -g pnpm` to install pnpm globally.
    2. **Install node modules**: Replaced `COPY package-lock.json package.json ./` with `COPY pnpm-lock.yaml ./` and `COPY package.json ./` to copy the pnpm lock file and package.json.
    3. **Install dependencies**: Replaced `RUN npm ci --include=dev` with `RUN pnpm install --frozen-lockfile` to install dependencies using pnpm.
    4. **Build application**: Replaced `RUN npx next build` with `RUN pnpm exec next build` to build the application using pnpm.
    5. **Prune development dependencies**: Replaced `RUN npm prune --omit=dev` with `RUN pnpm prune --prod` to remove development dependencies using pnpm.
    
    With these changes, your Dockerfile is now set up to use pnpm instead of npm.
    and this?
    <|editable_region_start|>
    CMD ["node", "server.js"]
    ```
    
    ### Explanation of Changes:
    1. **Install pnpm**: Added `RUN npm install -g pnpm` to install pnpm globally.
    2. **Install node modules**: Replaced `COPY package-lock.json package.json ./` with `COPY pnpm-lock.yaml ./` and `COPY package.json ./` to copy the pnpm lock file and package.json.
    3. **Install dependencies**: Replaced `RUN npm ci --include=dev` with `RUN pnpm install --frozen-lockfile` to install dependencies using pnpm.
    4. **Build application**: Replaced `RUN npx next build` with `RUN pnpm exec next build` to build the application using pnpm.
    5. **Prune development dependencies**: Replaced `RUN npm prune --omit=dev` with `RUN pnpm prune --prod` to remove development dependencies using pnpm.
    
    With these changes, your Dockerfile is now set up to use pnpm instead of npm.
    and this?
    <|editable_region_start|>
    CMD ["node", "server.js"]
    ```
    
    ### Explanation of Changes:
    1. **Install pnpm**: Added `RUN npm install -g pnpm` to install pnpm globally.
    2. **Install node modules**: Replaced `COPY package-lock.json package.json ./` with `COPY pnpm-lock.yaml ./` and `COPY package.json ./` to copy the pnpm lock file and package.json.
    3. **Install dependencies**: Replaced `RUN npm ci --include=dev` with `RUN pnpm install --frozen-lockfile` to install dependencies using pnpm.
    4. **Build application**: Replaced `RUN npx next build` with `RUN pnpm exec next build` to build the application using pnpm.
    5.
2025-02-22T12:28:22.041197Z [INFO] building git repository, `.git` path in the worktree: ".git"
2025-02-22T12:28:29.091803Z [INFO] Language server with id 4 sent unhandled notification eslint/status:
{
  "uri": "file:///Users/sakibulislam/Github/learn-framer-university/playwright.config.ts",
  "state": 1,
  "validationTime": 36
}
2025-02-22T12:28:29.125143Z [WARN] Get completion via vtsls failed: Request textDocument/completion failed with message: Reduce of empty array with no initial value
2025-02-22T12:28:29.208853Z [INFO] Language server with id 4 sent unhandled notification eslint/status:
{
  "uri": "file:///Users/sakibulislam/Github/learn-framer-university/playwright.config.ts",
  "state": 1,
  "validationTime": 11
}
2025-02-22T12:28:29.241738Z [WARN] Get completion via vtsls failed: Request textDocument/completion failed with message: Reduce of empty array with no initial value
2025-02-22T12:28:29.359479Z [INFO] Language server with id 4 sent unhandled notification eslint/status:
{
  "uri": "file:///Users/sakibulislam/Github/learn-framer-university/playwright.config.ts",
  "state": 1,
  "validationTime": 16
}
2025-02-22T12:28:29.39185Z [WARN] Get completion via vtsls failed: Request textDocument/completion failed with message: Reduce of empty array with no initial value
2025-02-22T12:28:29.426946Z [INFO] Language server with id 4 sent unhandled notification eslint/status:
{
  "uri": "file:///Users/sakibulislam/Github/learn-framer-university/playwright.config.ts",
  "state": 1,
  "validationTime": 17
}
2025-02-22T12:28:29.458435Z [WARN] Get completion via vtsls failed: Request textDocument/completion failed with message: Reduce of empty array with no initial value
2025-02-22T12:28:33.749463Z [ERROR] Failed to determine prettier path for buffer: empty metadata for initial path "/Users/sakibulislam/Github/learn-framer-university/package-lock.json"
2025-02-22T12:28:33.760073Z [ERROR] Failed to determine prettier ignore path for buffer: empty metadata for initial path "/Users/sakibulislam/Github/learn-framer-university/package-lock.json"
2025-02-22T12:28:34.075997Z [INFO] Language server with id 8 sent unhandled notification @/tailwindCSS/projectInitialized:
[
  null
]
2025-02-22T12:30:11.59077Z [INFO] Initializing default prettier with plugins {}
2025-02-22T12:31:46.772323Z [ERROR] edit prediction failed

Caused by:
    error predicting edits.
    Status: 500
    Body: Invalid output: <|editable_region_start|>
    
    /prompt Default 
    …
    …
    
    
    
    ```yml learn-framer-university/.github/workflows/ci.yml:1-28
    # Excerpt from: 
    name: CI
    
    on:
      push:
        branches:
          - main
      pull_request:
    
    concurrency:
      group: ci-${{ github.head_ref || github.ref }}
      cancel-in-progress: true
    
    env:
      CARGO_TERM_COLOR: always
      CARGO_MACHETE_VERSION: 0.7.0
      PNPM_VERSION: 10.4.1
      POSTGRES_VERSION: 16
    
    jobs:
      changed-files:
        name: Changed Files
        runs-on: ubuntu-24.04
    
        steps:
          - uses: actions/checkout@v4
            with:
              persist-credentials: false
    
          - name: Changed Files
            id: changed-files
            run: |
              echo "::set-output name=files::$(git diff --name-only HEAD^ HEAD | tr '\n' ' ')"
    ```
    
2025-02-22T12:31:48.434217Z [ERROR] edit prediction failed

Caused by:
    error predicting edits.
    Status: 500
    Body: Invalid output: <|editable_region_start|>
    
    /prompt Default 
    …
    …
    
    
    
    ```yml learn-framer-university/.github/workflows/ci.yml:1-28
    # Excerpt from: 
    name: CI
    
    on:
      push:
        branches:
          - main
      pull_request:
    
    concurrency:
      group: ci-${{ github.head_ref || github.ref }}
      cancel-in-progress: true
    
    env:
      CARGO_TERM_COLOR: always
      CARGO_MACHETE_VERSION: 0.7.0
      PNPM_VERSION: 10.4.1
      POSTGRES_VERSION: 16
    
    jobs:
      changed-files:
        name: Changed Files
        runs-on: ubuntu-24.04
    
        steps:
          - uses: actions/checkout@v4
            with:
              persist-credentials: false
    
          - name: Check Changed Files
            id: changed-files
            run: |
              echo "::set-output name=changed::$(git diff --name-only HEAD^ HEAD | wc -l)"
              echo "::set-output name=changed-files::$(git diff --name-only HEAD^ HEAD)"
    
    ```
    
2025-02-22T12:31:49.346993Z [ERROR] edit prediction failed

Caused by:
    error predicting edits.
    Status: 500
    Body: Invalid output: <|editable_region_start|>
    
    /prompt Default 
    …
    …
    
    
    
    ```yml learn-framer-university/.github/workflows/ci.yml:1-28
    # Excerpt from: 
    name: CI
    
    on:
      push:
        branches:
          - main
      pull_request:
    
    concurrency:
      group: ci-${{ github.head_ref || github.ref }}
      cancel-in-progress: true
    
    env:
      CARGO_TERM_COLOR: always
      CARGO_MACHETE_VERSION: 0.7.0
      PNPM_VERSION: 10.4.1
      POSTGRES_VERSION: 16
    
    jobs:
      changed-files:
        name: Changed Files
        runs-on: ubuntu-24.04
    
        steps:
          - uses: actions/checkout@v4
            with:
              persist-credentials: false
    
          - name: What is persist credentials?
            run: echo "Persist credentials is a feature that allows GitHub Actions to cache credentials between runs. It can be useful for speeding up workflows that require authentication, such as publishing packages or deploying to servers."
    
    ```
    
2025-02-22T12:31:49.631716Z [ERROR] edit prediction failed

Caused by:
    error predicting edits.
    Status: 500
    Body: Invalid output: <|editable_region_start|>
    
    /prompt Default 
    …
    …
    
    
    
    ```yml learn-framer-university/.github/workflows/ci.yml:1-28
    # Excerpt from: 
    name: CI
    
    on:
      push:
        branches:
          - main
      pull_request:
    
    concurrency:
      group: ci-${{ github.head_ref || github.ref }}
      cancel-in-progress: true
    
    env:
      CARGO_TERM_COLOR: always
      CARGO_MACHETE_VERSION: 0.7.0
      PNPM_VERSION: 10.4.1
      POSTGRES_VERSION: 16
    
    jobs:
      changed-files:
        name: Changed Files
        runs-on: ubuntu-24.04
    
        steps:
          - uses: actions/checkout@v4
            with:
              persist-credentials: false
    
          - name: Persist Credentials
            run: echo "persist-credentials=false" >> $GITHUB_ENV
    
    ```
    
2025-02-22T12:32:02.441828Z [ERROR] {
  "thread": "main",
  "payload": "begin <= end (54 <= 53) when slicing ` security by limiting the scope of the token's usage.\nwhy?`",
  "location_data": {
    "file": "/Users/administrator/actions-runner-4/_work/zed/zed/crates/rope/src/chunk.rs",
    "line": 158
  },
  "backtrace": [
    "backtrace::capture::Backtrace::create::h71b44b128423336c+10134660",
    "backtrace::capture::Backtrace::new::h743343b0418d06db+10134472",
    "zed::reliability::init_panic_hook::{{closure}}::hd0763ea60cdb075e+84622520",
    "std::panicking::rust_panic_with_hook::h09e8a656f11e82b2+65513696",
    "std::panicking::begin_panic_handler::{{closure}}::h1230eb3cc91b241c+65510792",
    "std::sys::backtrace::__rust_end_short_backtrace::hc3491307aceda2c2+65502896",
    "_rust_begin_unwind+65509980",
    "core::panicking::panic_fmt::ha4b80a05b9fff47a+94199480",
    "core::str::slice_error_fail_rt::h7cb1d51121ce60d3+16873784",
    "core::str::slice_error_fail::he72416048b197ca9+94200884",
    "rope::Cursor::summary::hb08703d5de36effc+20727192",
    "multi_buffer::MultiBufferSnapshot::text_summary_for_excerpt_offset_range::hfd4b32dc1688a2be+19491960",
    "multi_buffer::MultiBufferSnapshot::text_summary_for_range::hff7c41d293e84406+19480548",
    "editor::display_map::inlay_map::InlaySnapshot::text_summary_for_range::hb3f62a7915c71e34+22070376",
    "editor::display_map::fold_map::FoldSnapshot::text_summary_for_range::hed3cd76bb796927c+22038908",
    "editor::display_map::tab_map::TabSnapshot::text_summary_for_range::hc534a267453e3290+22076068",
    "editor::display_map::wrap_map::WrapSnapshot::text_summary_for_range::hb7caee3ec8b0b930+22109136",
    "editor::display_map::block_map::BlockMap::sync::hb5581f3c7e912380+21985972",
    "editor::display_map::block_map::BlockMap::read::h0d5602c0f5c4b9da+21976104",
    "editor::display_map::DisplayMap::snapshot::h96b7e9bb30985123+22132708",
    "editor::Editor::handle_input::{{closure}}::hfbd742ce0cf5a608+23385452",
    "editor::Editor::handle_input::h9852ece1a3f04038+23380720",
    "<editor::Editor as gpui::input::EntityInputHandler>::replace_text_in_range::h89a60c362d9eadd7+24233200",
    "<gpui::input::ElementInputHandler<V> as gpui::platform::InputHandler>::replace_text_in_range::hbd3ae9cac47b488f+21856832",
    "gpui::platform::PlatformInputHandler::replace_text_in_range::hf04850205a38cd0e+33448480",
    "gpui::platform::mac::window::insert_text::h00ac5c69d37cdd4c+33429724",
    "gpui::platform::mac::window::handle_key_event::h0f799f694db6b910+33413296",
    "<gpui::platform::mac::platform::MacPlatform as gpui::platform::Platform>::run::h181325e9ddacf5d5+33341912",
    "gpui::app::Application::run::hefc451e6fd13b21d+82326264",
    "zed::main::hbef7c747e1bb3d73+83266456",
    "std::sys::backtrace::__rust_begin_short_backtrace::h76f6ff3cfd258de3+84682804",
    "std::rt::lang_start::{{closure}}::h8d81c6332f22613a+84186160",
    "std::rt::lang_start_internal::hdd117cb81a316264+65444376",
    "_main+83276696"
  ],
  "app_version": "0.174.6",
  "app_commit_sha": "d5fde6645afe6356ef31611a76aff69a3e450260",
  "release_channel": "stable",
  "target": "aarch64-apple-darwin",
  "os_name": "macOS",
  "os_version": "15.2.0",
  "architecture": "aarch64",
  "panicked_on": 1740227522441,
  "system_id": "8064a90a-a880-448a-9448-8ff91b85d990",
  "installation_id": "244b9f48-ce95-4e54-a389-e7a413313f62",
  "session_id": "024de1c7-39cb-4341-b928-7674df6cb50e"
}
```
<!-- Click above this line and paste or drag-and-drop your log--></details>

--- a/crates/editor/src/display_map.rs
+++ b/crates/editor/src/display_map.rs
@@ -1925,7 +1925,7 @@ pub mod tests {
                 buffer.clone(),
                 font("Helvetica"),
                 font_size,
-                Some(px(72.0)),
+                None,
                 true,
                 1,
                 1,

--- a/crates/editor/src/display_map.rs
+++ b/crates/editor/src/display_map.rs
@@ -1964,8 +1964,8 @@ pub mod tests {
         });
         map.update(cx, |m, cx| assert_eq!(m.snapshot(cx).text(), "\n\n\na"));
 
-        dbg!(">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>");
-
+        // Regression test: updating the display map does not crash when a
+        // block is immediately followed by a multi-line inlay.
         buffer.update(cx, |buffer, cx| {
             buffer.edit([(1..1, "b")], None, cx);
         });

--- a/crates/editor/src/display_map.rs
+++ b/crates/editor/src/display_map.rs
@@ -1912,6 +1912,67 @@ pub mod tests {
     }
 
     #[gpui::test]
+    fn test_inlays_with_newlines_after_blocks(cx: &mut gpui::TestAppContext) {
+        cx.update(|cx| init_test(cx, |_| {}));
+
+        let buffer = cx.new(|cx| Buffer::local("a", cx));
+        let buffer = cx.new(|cx| MultiBuffer::singleton(buffer, cx));
+        let buffer_snapshot = buffer.read_with(cx, |buffer, cx| buffer.snapshot(cx));
+
+        let font_size = px(14.0);
+        let map = cx.new(|cx| {
+            DisplayMap::new(
+                buffer.clone(),
+                font("Helvetica"),
+                font_size,
+                Some(px(72.0)),
+                true,
+                1,
+                1,
+                1,
+                FoldPlaceholder::test(),
+                cx,
+            )
+        });
+
+        map.update(cx, |map, cx| {
+            map.insert_blocks(
+                [BlockProperties {
+                    placement: BlockPlacement::Above(
+                        buffer_snapshot.anchor_before(Point::new(0, 0)),
+                    ),
+                    height: 2,
+                    style: BlockStyle::Sticky,
+                    render: Arc::new(|_| div().into_any()),
+                    priority: 0,
+                }],
+                cx,
+            );
+        });
+        map.update(cx, |m, cx| assert_eq!(m.snapshot(cx).text(), "\n\na"));
+
+        map.update(cx, |map, cx| {
+            map.splice_inlays(
+                &[],
+                vec![Inlay {
+                    id: InlayId::InlineCompletion(0),
+                    position: buffer_snapshot.anchor_after(0),
+                    text: "\n".into(),
+                }],
+                cx,
+            );
+        });
+        map.update(cx, |m, cx| assert_eq!(m.snapshot(cx).text(), "\n\n\na"));
+
+        dbg!(">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>");
+
+        buffer.update(cx, |buffer, cx| {
+            buffer.edit([(1..1, "b")], None, cx);
+        });
+        map.update(cx, |m, cx| assert_eq!(m.snapshot(cx).text(), "\n\n\nab"));
+    }
+
+    #[gpui::test]
     async fn test_chunks(cx: &mut gpui::TestAppContext) {
         let text = r#"
             fn outer() {}

--- a/crates/editor/src/display_map/block_map.rs
+++ b/crates/editor/src/display_map/block_map.rs
@@ -638,10 +638,13 @@ impl BlockMap {
                 self.custom_blocks[start_block_ix..end_block_ix]
                     .iter()
                     .filter_map(|block| {
-                        Some((
-                            block.placement.to_wrap_row(wrap_snapshot)?,
-                            Block::Custom(block.clone()),
-                        ))
+                        let placement = block.placement.to_wrap_row(wrap_snapshot)?;
+                        if let BlockPlacement::Above(row) = placement {
+                            if row < new_start {
+                                return None;
+                            }
+                        }
+                        Some((placement, Block::Custom(block.clone())))
                     }),
             );
 

